central/src/main/java/ai/djl/serving/central/handler/ModelDownloadHandler.java

```diff
@@ -0,0 +1,118 @@
+        Map<Application, List<Artifact>> models = ModelZoo.listModels();
+        models.forEach(
+                (app, list) -> {
+                    list.forEach(
+                            artifact -> {
+                                if (artifact.getName().equals(modelName)){
+                                    for (Map.Entry<String, Artifact.Item> entry :
+                                            artifact.getFiles().entrySet()) {
+                                        URI fileUri = URI.create(entry.getValue().getUri());
+                                        URI baseUri = artifact.getMetadata().getRepositoryUri();
+                                        if (!fileUri.isAbsolute()) {
+                                            fileUri = BASE_URI.resolve(baseUri).resolve(fileUri);
+                                        }
+                                        try {
+                                            links.put(entry.getKey(),fileUri);
+                                        } catch(Exception e){
+                                            logger.info(String.valueOf(e));
+                                        }
+                                    }
+                                }});
+                });
+        return links;
+    }
+}
+
+
+/**
+ * A handler to handle download requests from the UI
+ * @author anfee1@morgan.edu
+ *
+ */
+public class ModelDownloadHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+
+    HttpRequestResponse jsonResponse;
+    public ModelDownloadHandler() { jsonResponse = new HttpRequestResponse(); }
+
+    /**
+     * handle the deployment request by forwarding the request to the serving-instance.
+     *
+     * @param ctx the context
+     * @param request the full request
+     */
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws IOException, ModelNotFoundException {
+        final Logger logger = LoggerFactory.getLogger(ModelDownloadHandler.class);
+        QueryStringDecoder decoder = new QueryStringDecoder(request.uri());
+        String modelName=NettyUtils.getParameter(decoder, "modelName", null);
+        CompletableFuture.supplyAsync(
+                () -> {
+                    try {
+                        if (modelName!=null) {
+                            logger.info(String.valueOf(ModelLink.linkFinder(modelName)));
+                            return ModelLink.linkFinder(modelName);
+                        } else {
+                            throw new BadRequestException("modelName is mandatory.");
+                        }
+
+                    } catch (IOException | ModelNotFoundException ex) {
+                        throw new IllegalArgumentException(ex.getMessage(), ex);
+                    }
+                })
+                .exceptionally((ex) -> Collections.emptyMap())
+                .thenAccept(linksMap -> jsonResponse.sendAsJson(ctx, request, linksMap));
```

Comment:
Does this means every web-client=browser need to have access to the url through firewall,proxy and eventuelly to the filesystem where central stores uploads ?
Cause this just returns a link and not downloads the files.

I just thinking about future improvements and just want top drop a few devOp thoughts...

1) every client need access to the url, in bigger enterprises with restricted network-policies it would be beneficially to grant access for the central server through all firewalls/proxies needed instead for every client. Same for serving instances running on cloud-nodes and need access
2) Authentication , if you plan to add role-based authentication and authorization in future it would be easier to allow central to check privileges and manages the download.  
3) upload artifacts using central, which is planned as I understood. when download runs through central there would be no need to grant access to file-system and central web-UI
4) using of central as Mirror in front of real-repositories. 
5) using central for caching of remote-model-repositories

just a few ideas, might be wrong if you want to scale out based on network-filesystem instead of scaling out central.

The ideas are just from a DevOp perspective